### ISSUE TITLE
install script doesn't work on alpine linux

### ISSUE THREAD

**Issue by semorrison** [Kha/elan#28]
Not sure if this is worth investigating, but I ran into it.

Steps to reproduce:
```
> docker run -it alpine
$ apk add curl
$ curl https://raw.githubusercontent.com/Kha/elan/master/elan-init.sh -sSf | sh -s -- -y
```

Produces output:
```
info: downloading installer
sh: /tmp/tmp.Jffhng/elan-init: not found
```

**Comment by semorrison** [comment]
I suspect the underlying issue is the difference in the behaviour of `mktemp` on alpine, described here: https://github.com/pires/docker-elasticsearch/issues/56#issuecomment-367033995.

**Comment by Kha** [comment]
It looks like that isn't the only issue: https://github.com/rust-lang/rustup/issues/354

**Comment by semorrison** [comment]
Curiously however the rustup install script `curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh` gets past this point on `alpine`.

**Comment by semorrison** [comment]
Switching from `alpine` to `frolvlad/alpine-glibc` allows `elan-init.sh` to work.

**Comment by Kha** [comment]
Yeah, looks like they fixed it after all in https://github.com/rust-lang/rustup/commit/151623408f839cb304104f43e7692f06eefc701d!

**Comment by Kha** [comment]
Would be nice to solve this when we invariably have to move on from Travis anyway...

**Comment by semorrison** [comment]
I made epsilon progress by copying `get_architecture()` from the latest `rustup-init.sh` into `elan-init.sh`. Now we get
```
info: downloading installer
curl: (22) The requested URL returned error: 404 
elan: command failed: curl -sSfL https://github.com/Kha/elan/releases/download/v0.10.3/elan-x86_64-unknown-linux-musl.tar.gz -o /tmp/tmp.gMpOMg/elan-init.tar.gz
```

**Comment by semorrison** [comment]
Is there any hope that https://github.com/Kha/elan/compare/master...semorrison:arch?expand=1 is all that's needed?

**Comment by Kha** [comment]
I have no idea!

**Comment by semorrison** [comment]
Okay, I managed to compile `elan` against `musl` (using [this](https://github.com/clux/muslrust)):

```
docker pull clux/muslrust
docker run -v $PWD:/volume --rm -t clux/muslrust cargo build
```

It seems to have produced plausible looking things in `elan/target/x86_64-unknown-linux-musl/debug`, but I don't really understand what happens in your release procedure to create the `.tar.gz` file that `elan-init.sh` downloads. Presumably I just grab everything from that directory.

**Comment by semorrison** [comment]
Ah, I see, you just
```
cd target/x86_64-unknown-linux-musl/debug && tar zcvf elan-x86_64-unknown-linux-musl.tar.gz elan-init
```
(which seems to be much larger than the `.tar.gz` files on your releases page?)

Then copying this over to a docker `alpine` image, everything works perfectly.

**Comment by Kha** [comment]
I'm pretty sure we package the release build. Anyway, I need this tested on Travis before I can merge it. I can give it a try later this week, at which point I'll probably combine it with the migration to GH Actions.

**Comment by Kha** [comment]
I should probably note, Lean 4 does not use static executables because it would break the new plugin support.
